Update Dockerfile to use Java 17

Nightly builds are using Dockerfile from this repository.

Since #2838 is merged, nightly builds fail with the error:
```
Welcome to Gradle 8.0.2!

Here are the highlights of this release:
 - Improvements to the Kotlin DSL
 - Fine-grained parallelism from the first build with configuration cache
 - Configurable Gradle user home cache cleanup

For more details see https://docs.gradle.org/8.0.2/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Build file '/tmp/build/841c1d7f/build.gradle' line: 2

* What went wrong:
An exception occurred applying plugin request [id: 'com.android.application', version: '8.1.4']
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in /usr/lib/jvm/java-11-openjdk-amd64
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

Dockerfile should be updated to use Java 17.

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM docker.io/ubuntu:18.04
+FROM docker.io/debian:12.4
 
 # Install Android Studio requirements
 # See https://developer.android.com/studio/install#linux
@@ -7,7 +7,7 @@ RUN apt-get update -y \
 wget \
 curl \
 unzip \
-openjdk-11-jre \
+openjdk-17-jre \
 file \
 build-essential \
 && rm -rf /var/lib/apt/lists/*

